app/containers/Demo/index.js

```diff
@@ -0,0 +1,210 @@
+/**
+ *
+ * Demo
+ *
+ */
+
+import React, { memo } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { injectIntl } from 'react-intl';
+import { Input, Card, Skeleton } from 'antd';
+import styled from 'styled-components';
+import debounce from 'lodash/debounce';
+import { Link } from 'react-router-dom';
+import { createStructuredSelector } from 'reselect';
+import { compose } from 'redux';
+import { useInjectSaga } from '@utils/injectSaga';
+import makeSelectDemo, { selectError, selectSearchText, selectSongs, selectLoading } from './selectors';
+import { demoCreators } from './reducer';
+import saga from './saga';
+
+// styled components
+
+const CustomText = styled.small`
+  font-size: ${props => props.size}px;
+  color: ${props => (props.color ? props.color : '#333')};
+  margin-bottom: 5px;
+  display: block;
+  width: 280px;
+  text-overflow: ellipsis;
+  overflow: hidden;
+  white-space: nowrap;
+  a {
+    color: inherit;
+  }
+`;
+
+const BackLink = styled.div`
+  a {
+    color: #333;
+    font-weight: bold;
+    font-size: 15px;
+    display: block;
+    margin-bottom: 30px;
+
+    &:hover {
+      color: orange;
+    }
+  }
+`;
+
+const CustomCard = styled(Card)`
+  && {
+    display: flex;
+    flex-direction: column;
+    .inner-wrapper {
+      display: flex;
+      flex: 1;
+      flex-direction: column;
+      justify-content: space-between;
+      height: 100%;
+    }
+    .ant-card-cover {
+      height: 250px;
+    }
+    img {
+      width: 100%;
+      height: 100%;
+      object-fit: cover;
+    }
+    .audio-control {
+    }
+    audio {
+      width: 100%;
+      margin-top: 20px;
+    }
+  }
+`;
+
+const CustomInput = styled(Input)`
+  && {
+    padding: 12px;
+    box-shadow: 3px 3px 5px rgba(0, 0, 0, 0.04);
+    background-color: inherit;
+    border: 1px solid #e8e8e8;
+    transition: all 0.2s;
+    &:hover {
+      box-shadow: 3px 3px 5px rgba(0, 0, 0, 0.04);
+      border: 1px solid #e8e8e8;
+    }
+    &:focus {
+      box-shadow: 3px 3px 10px rgba(0, 0, 0, 0.04);
+      border: 1px solid #e8e8e8;
+      border-bottom: 1px solid orange;
+    }
+  }
+`;
+
+const FlexWrapper = styled.div`
+  display: flex;
+`;
+
+const CenteredDiv = styled.div`
+  max-width: 1100px;
+  margin: 30px auto;
+  padding: 0 20px;
+`;
+const GridCard = styled.div`
+  display: grid;
+  margin-top: 30px;
+  grid-template-columns: repeat(auto-fit, minmax(300px, 1fr));
+  gap: 20px;
+`;
+
+const renderSongs = (songs, loading) => {
```

Comment:
nice, like what you've done here by moving this outside the `Demo` component. 

Let's create another component for this and move this there